spec/integration/interactive_mode_spec.rb

```diff
@@ -0,0 +1,99 @@
+# frozen_string_literal: true
+
+require 'codeowners/checker'
+
+RSpec.describe 'Interactive mode' do
+  it 'runs with no-issues reporting' do
+    start(
```

Comment:
I really like the way you implemented the tests for interactive mode 👏 

Two thoughts:
* Maybe we can improve the name and use something like `invoke_cli` or `start_interactive_mode` or something that is less generic than `start`. What do you think?

Secondly, I was wondering if we could make this an object instead. My idea was something like:
```ruby
  subject do # or let(:interactive_mode) do end if that is the preferred style
    InteractiveMode.new(codeowners: ... , owners: ..., file_tree: ...)
  end
  it ... do
    expect(subject.run).to ....
  end
```

This way, we would have an object work with, instead of calling a helper method that yields. This might be easier for debugging failing tests. It also seems to more Ruby-esk.
That said, there is nothing wrong with your approach and I am fine either way. Just want to make sure we keep it readable and simple.

Reply:
yes, I was thinking about implementing it with the class too, as it feels a more extendable way to implement this feature. Just from the start, I didn't know that testing this would be pretty tricker and will envolved more than a couple of helpers, so sure I'll convert these helpers into a more OO way.